ddtrace/contrib/unittest/patch.py

```diff
@@ -0,0 +1,183 @@
+import inspect
+import os
+import unittest
+
+import ddtrace
+from ddtrace import config
+from ddtrace.constants import SPAN_KIND
+from ddtrace.contrib.unittest.constants import COMPONENT_VALUE
+from ddtrace.contrib.unittest.constants import FRAMEWORK
+from ddtrace.contrib.unittest.constants import KIND
+from ddtrace.ext import SpanTypes
+from ddtrace.ext import test
+from ddtrace.internal.ci_visibility import CIVisibility as _CIVisibility
+from ddtrace.internal.ci_visibility.constants import EVENT_TYPE as _EVENT_TYPE
+from ddtrace.internal.constants import COMPONENT
+from ddtrace.internal.utils.wrappers import unwrap as _u
+from ddtrace.vendor import wrapt
+
+
+# unittest default settings
+config._add(
+    "unittest",
+    dict(
+        _default_service="unittest",
+        operation_name=os.getenv("DD_UNITTEST_OPERATION_NAME", default="unittest.test"),
+    ),
+)
+
+
+def _set_tracer(tracer):
+    unittest._datadog_tracer = tracer
+
+
+def _store_span(item, span):
+    """Store span at `unittest` instance."""
+    item._datadog_span = span
+
+
+def _extract_span(item):
+    """Extract span from `unittest` instance."""
+    return getattr(item, "_datadog_span", None)
+
+
+def _extract_test_method_name(item):
+    """Extract test method name from `unittest` instance."""
+    return getattr(item, "_testMethodName", None)
+
+
+def _extract_suite_name_from_test_method(item):
+    item_type = type(item)
+    return getattr(item_type, "__name__", None)
+
+
+def _extract_class_hierarchy_name(item):
+    item_type = type(item)
+    return getattr(item_type, "__name__", None)
+
+
+def _extract_module_name_from_test_method(item):
+    return getattr(item, "__module__", None)
+
+
+def _extract_test_skip_reason(args):
+    return args[1]
+
+
+def _extract_test_file_name(item):
+    return os.path.basename(inspect.getfile(item.__class__))
+
+
+def is_unittest_support_enabled():
+    return unittest and getattr(unittest, "_datadog_patch", False) and _CIVisibility.enabled
+
+
+def _is_valid_result(instance, args):
+    return instance and type(instance) == unittest.runner.TextTestResult and args
+
+
+def patch():
+    """
+    Patch the instrumented methods from unittest
+    """
+    if (
+        not config._ci_visibility_unittest_enabled
+        or getattr(unittest, "_datadog_patch", False)
+        or _CIVisibility.enabled
+    ):
+        return
+
+    _CIVisibility.enable(config=ddtrace.config.unittest)
+
+    unittest._datadog_patch = True
```

Comment:
Should this attribute be set using `setattr`? From a quick glance this should raise an AttributeError

Reply:
It was set using `setattr` but the linter was raising an error about it. See: [CI workflow](https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/44127/workflows/d4a371fc-0f36-4dab-8f89-86f91ae42cf2/jobs/2911016/parallel-runs/0/steps/0-104)

Reply:
When do setting an attribute cause an `AttributeError` exception that also doesn't happen with `setattr()` ?

Doesn't it behave the same?
```
>>> my_none = None
>>> my_none.blah = True
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'blah'
>>> setattr(my_none, "blah", True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'blah'
```
vs
```
>>> my_func = lambda: None
>>> my_func.blah = True
>>> setattr(my_func, "blah", False)
```